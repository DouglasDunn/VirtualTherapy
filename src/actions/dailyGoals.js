import database from '../firebase/firebase';

// ADD_DAILY_GOAL
export const addDailyGoal = (dailyGoal) => ({
  type: 'ADD_DAILY_GOAL',
  dailyGoal
});

export const startAddDailyGoal = (dailyGoalData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      goal = '',
      completed = false
    } = dailyGoalData;
    const dailyGoal = { goal, completed };

    return database.ref(`users/${uid}/dailyGoals`).push(dailyGoal).then((ref) => {
      dispatch(addDailyGoal({
        id: ref.key,
        ...dailyGoal
      }));
    });
  };
};

// SET_EXPENSES
export const setDailyGoals = (dailyGoals) => ({
  type: 'SET_DAILY_GOALS',
  dailyGoals
});

export const startSetDailyGoals = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/dailyGoals`).once('value').then((snapshot) => {
      const dailyGoals = [];

      snapshot.forEach((childSnapshot) => {
        dailyGoals.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setDailyGoals(dailyGoals));
    });
  };
};
