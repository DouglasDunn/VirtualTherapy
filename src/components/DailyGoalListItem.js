import React from 'react';
import { Link } from 'react-router-dom';

const DailyGoalListItem = ({ id, goal }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{goal}</h3>
    </div>
  </Link>
);

export default DailyGoalListItem;
