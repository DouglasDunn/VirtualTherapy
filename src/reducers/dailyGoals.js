// DailyGoals Reducer

const dailyGoalsReducerDefaultState = [];

export default (state = dailyGoalsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_DAILY_GOAL':
      return [
        ...state,
        action.dailyGoal
      ];
    case 'SET_DAILY_GOALS':
      return action.dailyGoals;
    default:
      return state;
  }
};
