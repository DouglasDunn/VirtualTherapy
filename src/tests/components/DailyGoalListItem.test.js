import React from 'react';
import { shallow } from 'enzyme';
import dailyGoals from '../fixtures/dailyGoals';
import DailyGoalListItem from '../../components/DailyGoalListItem';

test('should render DailyGoalListItem correctly', () => {
  const wrapper = shallow(<DailyGoalListItem {...dailyGoals[0]} />);
  expect(wrapper).toMatchSnapshot();
});
