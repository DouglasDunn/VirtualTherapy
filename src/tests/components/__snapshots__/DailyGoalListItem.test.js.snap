// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DailyGoalListItem correctly 1`] = `
<Link
  className="list-item"
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Run
    </h3>
  </div>
</Link>
`;
